python/ray/data/_internal/execution/autoscaler/default_autoscaler.py

```diff
@@ -0,0 +1,192 @@
+import math
+import time
+from typing import TYPE_CHECKING, Dict
+
+from .autoscaler import Autoscaler
+from .autoscaling_actor_pool import AutoscalingActorPool
+from ray.data._internal.execution.autoscaling_requester import (
+    get_or_create_autoscaling_requester_actor,
+)
+from ray.data._internal.execution.interfaces.execution_options import ExecutionResources
+
+if TYPE_CHECKING:
+    from ray.data._internal.execution.interfaces import PhysicalOperator
+    from ray.data._internal.execution.resource_manager import ResourceManager
+    from ray.data._internal.execution.streaming_executor_state import (
+        OpSchedulingStatus,
+        OpState,
+        SchedulingDecision,
+        Topology,
+    )
+
+
+class DefaultAutoscaler(Autoscaler):
+
+    # Default threshold of actor pool utilization to trigger scaling up.
+    DEFAULT_ACTOR_POOL_SCALING_UP_THRESHOLD: float = 0.8
+    # Default threshold of actor pool utilization to trigger scaling down.
+    DEFAULT_ACTOR_POOL_SCALING_DOWN_THRESHOLD: float = 0.5
+
+    # Min number of seconds between two autoscaling requests.
+    MIN_GAP_BETWEEN_AUTOSCALING_REQUESTS = 20
+
+    def __init__(
+        self,
+        topology: "Topology",
+        resource_manager: "ResourceManager",
+        execution_id: str,
+        actor_pool_scaling_up_threshold: float = DEFAULT_ACTOR_POOL_SCALING_UP_THRESHOLD,  # noqa: E501
+        actor_pool_scaling_down_threshold: float = DEFAULT_ACTOR_POOL_SCALING_DOWN_THRESHOLD,  # noqa: E501
+    ):
+        self._actor_pool_scaling_up_threshold = actor_pool_scaling_up_threshold
+        self._actor_pool_scaling_down_threshold = actor_pool_scaling_down_threshold
+        # Last time when a request was sent to Ray's autoscaler.
+        self._last_request_time = 0
+        super().__init__(topology, resource_manager, execution_id)
+
+    def try_trigger_scaling(self, scheduling_decision: "SchedulingDecision"):
+        self._try_scale_up_cluster(scheduling_decision)
+        self._try_scale_up_or_down_actor_pool(scheduling_decision)
+
+    def _actor_pool_util(self, actor_pool: AutoscalingActorPool):
```

Comment:
Nit: Could we rename this method to be more descriptive? I wasn't able to infer what the method does from the name `_actor_pool_util`

Reply:
will update it to `_calculate_actor_pool_util`

Reply:
Oh, does "util" refer to "utilization" in this context? Thought it meant "util" as in "utility"

Reply:
yeah, I didn't realize "util" is also an common abbreviation for "utility"